### ISSUE TITLE
Build `bolinas-dna/evals_dart_task3` — DART-Eval Task 3 cell-type-specific peak dataset

### ISSUE THREAD

**Issue by gonzalobenegas** [Open-Athena/marin-dna#293]
🤖 Agent-drafted design/tracking issue. Decisions made with @gonzalobenegas up front: (1) build it as a new dataset in `snakemake/evals` and upload to HF as `bolinas-dna/evals_dart_task3`; (2) **minimal schema** — coordinates + cell-type label, split by file; (3) **scope is the dataset artifact only** — embedding UMAP and DART-Eval's quantitative metrics are out of scope here, tracked as separate follow-ups.

## Goal

Bring **DART-Eval Task 3 — "Discriminating Cell-Type-Specific Elements"** ([Patel et al., NeurIPS 2024 D&B](https://github.com/kundajelab/DART-Eval)) into our evals pipeline as a clean, well-provenanced HF dataset: `bolinas-dna/evals_dart_task3`. Producing and uploading that dataset is the whole deliverable.

**Why we want it.** It is the **interpretation / embedding** counterpart to the DART-Eval **variant-effect** datasets we already ship (`caqtl`/`dsqtl`, #139 / #236; VEP protocol in #262). As a standard HF dataset it unblocks embedding-based analyses on cell-type-specific accessible elements — e.g. a cell-type embedding UMAP (the analogue of the #246 GPN-Star functional-region UMAP) and DART-Eval's own Task 3 metrics (zero-shot clustering AMI, probing accuracy). **Those analyses are deliberately out of scope here** and will be separate issues; this one just builds the artifact.

## What Task 3 is

DART-Eval takes ATAC-seq from **5 cell lines** (GM12878, H1ESC, HEPG2, IMR90, K562), builds a consensus peakset, runs DESeq2 for differential accessibility, and selects the **top 5,000 differentially-accessible peaks per cell type**. Each peak is a **500 bp** window (±250 bp around the ATAC summit), GRCh38, labeled by the cell type it is differentially accessible in. After de-duplicating peaks selected by more than one cell type, **24,991 windows** remain — exactly the file DART-Eval's own embedders consume (`peaks_by_cell_label_unique_dataloader_format.tsv`).

## Source

DART-Eval Synapse project [`syn60581042`](https://www.synapse.org/Synapse:syn60581042). We consume their pre-computed, deduped per-cell-label window file directly — **no re-running DESeq2**. Download is via a Synapse Personal Access Token over plain HTTP, the exact mechanism `caqtl`/`dsqtl` already use (`dart_eval_download` rule, `SYNAPSE_AUTH_TOKEN`). The specific file entity ID is looked up once from the Task 3 folder and pinned in config, mirroring how `caqtl_synapse_id` / `dsqtl_synapse_id` are pinned today.

## Dataset schema — `bolinas-dna/evals_dart_task3`

500 bp **interval** windows, not variants — so no ref/alt, no consequence/VEP annotation, no matching/subsampling. **Split by file** (HF convention): `train.parquet` / `validation.parquet` / `test.parquet`; no in-row split column.

| Column | Type | Notes |
|---|---|---|
| `chrom` | str | Ensembl-style (no `chr` prefix), GRCh38 |
| `start` | int64 | 0-based, inclusive |
| `end` | int64 | 0-based, exclusive; `end - start == 500` for every row (full consensus peak; window midpoint = ATAC summit) |
| `label` | str | Cell type: `GM12878` / `H1ESC` / `HEPG2` / `IMR90` / `K562` |

We store the **full 500 bp peak**, never pre-cropped to a model's context window — so downstream embedding / pooling / context choices stay open. DESeq statistics (log2FC / padj) are intentionally **not** carried (minimal schema); they remain in DART-Eval's `top_5000_deseq_peaks.tsv` if ever needed.

## Splits — DART-Eval canonical, one file per split

Shipped as three files using HF's standard split filenames, with windows routed by chromosome per DART-Eval's Task 3 training scripts (e.g. `experiments/task_3_peak_classification/train/hyenadna.py`):

| File | Chromosomes |
|---|---|
| `train.parquet` | 1, 2, 3, 4, 7, 8, 9, 11, 12, 13, 15, 16, 17, 19, X, Y |
| `validation.parquet` | 6, 21 |
| `test.parquet` | 5, 10, 14, 18, 20, 22 |

A **3-way** chromosome holdout, distinct from the evals pipeline's default odd/even **2-way** `SPLIT_CHROMS`, kept identical to DART-Eval so any future Task 3 numbers stay comparable to theirs. Per-split row counts follow the peaks' genomic distribution (not equal sizes).

## Implementation outline

Self-contained `dart_task3` module mirroring the `caqtl`/`dsqtl` Synapse→HF pattern but **interval-native** (skips the variant machinery — `check_ref_alt`, consequence/TSS/exon annotation, `materialize_eval_harness_dataset`, matching/QC):

1. **Parser** — `src/marin_dna/pipelines/evals/dart_task3.py`: `parse_dart_task3()` reads the dataloader-format TSV → `chrom` (strip `chr`) / `start` / `end` / `label`; asserts 5 cell-type labels, `end - start == 500`, ~25k rows, no null labels, chroms ⊆ canonical set. `split_by_chrom()` routes rows to train / validation / test from the canonical table and asserts full coverage. Unit-tested in `tests/`.
2. **Rules** — `snakemake/evals/workflow/rules/dart_task3.smk`: `dart_task3_download` (reuse the Synapse-PAT `curl` shell from `dart_eval_download`), `dart_task3_dataset` (parse + route → `results/dataset/dart_task3/{train,validation,test}.parquet`), `dart_task3_upload` (`HfApi` → `bolinas-dna/evals_dart_task3`: 3 split files + README). Dedicated rules rather than the variant-centric generic `split_dataset_by_chrom` / `hf_upload` (3 splits, interval card).
3. **Config** — `snakemake/evals/config/config.yaml`: add `task3_synapse_id` under the existing `dart_eval:` block.
4. **Dataset card** — interval / cell-type README: commit-pinned permalink to the pipeline, DART-Eval + ENCODE + cell-line provenance/credit, tags `biology, genomics, dna`. (New small renderer; the existing `render_dart_eval` is variant-specific.)
5. **READMEs** — `snakemake/evals/README.md` dataset table + a Task-3 subsection.

**Reproducibility / licensing note.** Download needs a free Synapse account (PAT as `SYNAPSE_AUTH_TOKEN`), same as `caqtl`/`dsqtl`. DART-Eval's repo carries no explicit data license; the underlying ATAC-seq is ENCODE (freely redistributable under the ENCODE data-use policy). Confirm terms and credit DART-Eval / ENCODE in the card before the HF push.

## Related

- #139 DART-Eval (parent; closed) · #236 Task 5 / ARSENAL VEP · #262 caQTL/dsQTL VEP protocol — the **variant-effect** side of DART-Eval
- #246 GPN-Star Fig 4 embedding UMAP — the embedding-UMAP infrastructure a future cell-type-UMAP analysis would reuse
- DART-Eval: Patel et al., NeurIPS 2024 Datasets & Benchmarks — https://github.com/kundajelab/DART-Eval

**Comment by gonzalobenegas** [comment]
🤖 Pipeline implemented in #294 — `evals_dart_task3` as an interval dataset (3-way split, minimal `chrom`/`start`/`end`/`label` schema, full 500 bp peak stored), self-contained in `snakemake/evals` and skipping the variant machinery. 16 new unit tests + snakemake dry-run green.

**One blocker before the HF artifact can be built/uploaded:** `config.dart_eval.task3_synapse_id` is a placeholder. The real Synapse entity ID for `peaks_by_cell_label_unique_dataloader_format.tsv` (under DART-Eval project [`syn60581042`](https://www.synapse.org/Synapse:syn60581042)) needs an authenticated Synapse-folder lookup. Once it's set (and `SYNAPSE_AUTH_TOKEN` exported):

```bash
cd snakemake/evals
uv run snakemake dart_task3                       # build the 3 local parquets
uv run snakemake results/dart_task3/upload.done   # upload to bolinas-dna/evals_dart_task3
```

Leaving this issue open until the dataset is actually on HF.

**Comment by gonzalobenegas** [comment]
🤖 Update — found the source file and validated against real data (now that I have Synapse access).

**Correction:** the file #293's body named (`peaks_by_cell_label_unique_dataloader_format.tsv`) is DART-Eval's **full** cell-type-specific set — **216,746** windows, for the supervised probing/fine-tuning task. The **~24,991 / top-5,000** figure actually belongs to a *different* file, `top_5000_deseq_peaks.tsv` ([`syn62161401`](https://www.synapse.org/Synapse:syn62161401)) — their balanced **clustering / UMAP subset** (exactly 5,000 per cell type → **25,000** windows, unique coords).

Per the UMAP motivation, `evals_dart_task3` uses the **top-5,000 subset** (#294 updated accordingly). Parser + asserts + canonical 3-way split validated 1:1 on the real file → train 17,965 / validation 1,958 / test 5,077 = 25,000. Build + HF upload pending a quick review of the generated card.

**Comment by gonzalobenegas** [comment]
🤖 ✅ **Done — [bolinas-dna/evals_dart_task3](https://huggingface.co/datasets/bolinas-dna/evals_dart_task3) is live.**

Built through the pipeline (`snakemake results/dart_task3/upload.done`, single atomic commit) and hand-verified on HF:

| File | Windows | per cell type |
|---|---:|---|
| `train.parquet` | 17,965 | ~3,500 each |
| `validation.parquet` | 1,958 | 318–557 |
| `test.parquet` | 5,077 | 802–1,140 |
| **total** | **25,000** | 5,000 each (balanced) |

Schema `chrom / start / end / label`, 500 bp, GRCh38, 0-based; card has the full per-cell-type × split table + provenance + ENCODE/DART-Eval credit.

Implementation in #294 (review-clean). Safe to close once #294 merges — and re-running the upload post-merge will refresh the card's provenance permalink from the branch commit to a `main` SHA.
